src/main/java/sirius/biz/storage/layer3/VirtualFile.java

```diff
@@ -0,0 +1,794 @@
+        }
+    }
+
+    /**
+     * Returns the last modification timestamp as {@link LocalDateTime}.
+     *
+     * @return the last modification timestamp
+     */
+    public LocalDateTime lastModifiedDate() {
+        try {
+            return lastModifiedSupplier == null ?
+                   null :
+                   Instant.ofEpochMilli(lastModifiedSupplier.apply(this))
+                          .atZone(ZoneId.systemDefault())
+                          .toLocalDateTime();
+        } catch (Exception e) {
+            throw handleErrorInCallback(e, "lastModifiedSupplier");
+        }
+    }
+
+    /**
+     * Returns the size of this file.
+     *
+     * @return the size in bytes
+     */
+    public long size() {
+        try {
+            return sizeSupplier == null ? 0 : sizeSupplier.apply(this);
+        } catch (Exception e) {
+            throw handleErrorInCallback(e, "sizeSupplier");
+        }
+    }
+
+    /**
+     * Determines if this file can (probably) be deleted.
+     *
+     * @return <tt>true</tt> if this file can be delete or <tt>false</tt> otherwise
+     */
+    public boolean canDelete() {
+        try {
+            if (canDeleteHandler != null) {
+                return canDeleteHandler.test(this);
+            }
+
+            return deleteHandler != null;
+        } catch (Exception e) {
+            throw handleErrorInCallback(e, "canDeleteHandler");
+        }
+    }
+
+    /**
+     * Tries to delete this file.
+     *
+     * @return <tt>true</tt> if the operation was successful, <tt>false</tt> otherwise
+     */
+    public boolean tryDelete() {
+        try {
+            if (deleteHandler == null) {
+                return false;
+            }
+
+            return deleteHandler.apply(this);
+        } catch (Exception e) {
+            throw handleErrorInCallback(e, "deleteHandler");
+        }
+    }
+
+    /**
+     * Deletes this file.
+     *
+     * @throws HandledException if the file cannot be deleted
+     */
+    public void delete() {
+        if (!tryDelete()) {
+            throw Exceptions.createHandled().withNLSKey("VirtualFile.cannotDelete").set("file", path()).handle();
+        }
+    }
+
+    /**
+     * Determines if this file can (probably) be renamed.
+     *
+     * @return <tt>true</tt> if this file can be renamed or <tt>false</tt> otherwise
+     */
+    public boolean canRename() {
+        try {
+            if (canRenameHandler != null) {
+                return canRenameHandler.test(this);
+            }
+
+            return renameHandler != null;
+        } catch (Exception e) {
+            throw handleErrorInCallback(e, "canRenameHandler");
+        }
+    }
+
+    /**
+     * Tries to rename this file.
+     *
+     * @return <tt>true</tt> if the operation was successful, <tt>false</tt> otherwise
+     */
+    public boolean tryRename(String newName) {
+        try {
+            if (renameHandler == null) {
```

Comment:
I think inverting this if check would improve readability/flow and woudl be consistent to the other methods in this class